src/v/pandaproxy/server.h

```diff
@@ -129,13 +132,32 @@ class server {
     ss::future<> stop();
 
 private:
+    class probe {
+    public:
+        probe(context_t& ctx, const ss::sstring& group_name);
+        probe(const probe&) = delete;
+        probe& operator=(const probe&) = delete;
+        probe(probe&&) = delete;
+        probe& operator=(probe&&) = delete;
+        ~probe() = default;
+
+    private:
+        void setup_metrics();
+
+        context_t& _ctx;
+        const ss::sstring& _group_name;
+        metrics::internal_metric_groups _metrics;
+        metrics::public_metric_groups _public_metrics;
+    };
+
     ss::httpd::http_server _server;
     ss::sstring _public_metrics_group_name;
     ss::gate _pending_reqs;
     ss::httpd::api_registry_builder20 _api20;
     bool _has_routes;
     context_t& _ctx;
     json::serialization_format _exceptional_mime_type;
+    probe _probe;
```

Comment:
`probe` could be forward declared, held by `std::unique_ptr`, started and stopped in `server::start/stop`, and the definition along with the metrics include all moved to the implementation file.